### ISSUE TITLE
Extension Hoxx VPN Proxy not working

### ISSUE THREAD

**Issue by Pilford** [ghacksuserjs/ghacks-user.js#295]
I was lucky to find perfect replacements for all my legacy extensions, except [Hoxx VPN Proxy](https://addons.mozilla.org/en-US/firefox/addon/hoxx-vpn-proxy/). I tried several things in the user.js file but to no avail. Which entry or entries do I have to change in order to make it work?

**Comment by Thorin-Oakenpants** [comment]
I'm not sure .. the easiest way is to use the halving method to troubleshoot. https://github.com/ghacksuserjs/ghacks-user.js/wiki/1.4-Troubleshooting.

Create a new profile, add the Hoxx VPN Proxy, make sure it works, clone the profile and add half the user.js and see what happens .. etc

**Comment by Pilford** [comment]
> but all I got was the toolbar button which when clicked just showed a progress wheel and "attempting to connect" .. is this what you get?

Yep, this is exactly what I get - an endless attempt to connect.

**Comment by Thorin-Oakenpants** [comment]
Did you see my "update" in my last post? I get the same as you with my actual FF which is the user.js with extensions etc. And yet in a brand new profile I cannot fault the user.js. Maybe it's got a hangup only on the setting language part/setup.

I'll test again, this time with a new profile + user.js and THEN I'll install Hoxx .. brb

**Comment by Thorin-Oakenpants** [comment]
^^ Yup, that's it. Something is blocking it from bringing up the language settings when installed. After that is set, the user.js is not blocking it (I assume).

1. New profile
2. add user.js and restart
3. add hoxxy - cannot get to select language screen
4. remove user.js
5. start resetting prefs and restarting to find the culprit (its not FPI or RFP)

**Comment by Pilford** [comment]
I've read all your updates and edits ;-) With a fresh profile I can indeed enter my account info, but it still seems to hang. No matter what site I want to visit I see “Performing a TLS handshake to”, and then soon after that “Problem loading page”.

**Comment by Thorin-Oakenpants** [comment]
I looked in the console, it's `dom.workers.enabled` (2301)

disable that in about config and the language setting will come up. I think you *might* be able to then disable workers again - up to you to test that out :)

**Comment by Thorin-Oakenpants** [comment]
try the tls settings, try service workers .. u can dooooo eeet :)

**Comment by Pilford** [comment]
I played a bit with the ServiceWorker entries but still no go. In the browser console I see:

1512057667700
Toolkit.Telemetry
WARN	TelemetryEnvironment::setExperimentActive - the experiment type was truncated.
Error: WebExtension context not found!

**Comment by Thorin-Oakenpants** [comment]
I saw something like that as well - sorry, destroyed my test profiles. It's probably a red herring. Clear the console and then pop out the extension button thingy. I have to scoot, maybe @earthling can nut it out a bit

**Comment by earthlng** [comment]
If it's something with TLS then I'd start with resetting these: about:config?filter=security.

**Comment by 2glops** [comment]
@Pilford 

If you use Linux, it could be related to the security sandbox level:
http://www.morbo.org/2017/11/linux-sandboxing-improvements-in.html

**Comment by Thorin-Oakenpants** [comment]
Any update on this?

**Comment by Pilford** [comment]
I’m too clumsy to figure out which setting I have to change in order to make it work, so I threw in the towel. For now I use a temp profile without the user.js file when I want to see some program on e.g. BBC iPlayer or TenPlay.

**Comment by Thorin-Oakenpants** [comment]
^^ OK. If I get time I'll set up an account and give it a whirl

**Comment by Thorin-Oakenpants** [comment]
@Pilford I applied the entire user.js and everything works.. but

- I installed Hoxxy with no user.js, which meant I was able to set my language - which we established above was a problem with `dom.workers.enabled`=false (2301) - so I guess you can flip that to true to set your language, and then flip it back to false
- I did set `security.OCSP.require`=false (1212) because it blocked my testing page - I actually have this as false in my main FF overrides because it causes too much breakage for me
- I did see some times a bit of a slowdown with TLS handshakes, but nothing that BROKE anything - this is likely the SSL session ID tickets - see pref `1203` `security.ssl.disable_session_identifiers` if you want to flip this

I cannot break Hoxxy with the current user.js

**Comment by Thorin-Oakenpants** [comment]
Also, as @2glops asked ... @Pilford what is your OS and are you running any extensions. Also, are you using FF57?

**Comment by Pilford** [comment]
You're a genius, Pants, and you know it! I changed "security.OCSP.require" from "true" to "false" and everything works like a charm. I tested it on both FF57 and Nightly.

I'm on Win7 and I use the following extensions:

CanvasBlocker
Context Search WebExtension
Decentraleyes
Google search link fix
HTTPS Everywhere
I don't care about cookies
Neat URL
Quick Google Translate
Save Page WE
Smart Referer
uBlock Origin
